Source/PINDiskCache.m

```diff
@@ -830,7 +866,7 @@ - (void)enumerateObjectsWithBlockAsync:(PINDiskCacheFileURLEnumerationBlock)bloc
 - (void)synchronouslyLockFileAccessWhileExecutingBlock:(PINCacheBlock)block
 {
     if (block) {
-        [self lock];
+        [self lockUntilWritable];
             block(self);
         [self unlock];
     }
```

Comment:
Should we also modify `containsObjectForKey:`?

Reply:
Good catch!